src/main/scala/com/kinja/amqp/UnconfirmedMessageRepeater.scala

```diff
@@ -98,11 +110,15 @@ class UnconfirmedMessageRepeater(
 		transactional: TransactionalMessageStore,
 		republishTimeout: FiniteDuration
 	)(implicit ec: ExecutionContext): Unit = {
-		msgs.map { msg =>
+		msgs.foreach { msg =>
 			val result = Try(Await.result(producer.publish(msg.routingKey, Json.parse(msg.message)), republishTimeout))
 			result match {
 				case Success(_) => deleteMessageAndMatchingConfirm(msg, confs, transactional)
-				case Failure(ex) => logger.warn(s"""Couldn't resend message: $msg, ${ex.getMessage}""")
+				case Failure(ex: TimeoutException) =>
+					deleteMessageAndMatchingConfirm(msg, confs, transactional)
```

Comment:
I guess this way the delete itself won't work either, but it's a tricky
question.
What is sure that without this the messeges get duplicated when rabbit is
unreachable, and unfortunately we get back this exception, not a more
specific one.
I'll check out what can we do about it tomorrow.

On Monday, April 27, 2015, Levente Molnar <notifications@github.com> wrote:

> In src/main/scala/com/kinja/amqp/UnconfirmedMessageRepeater.scala
> <https://github.com/gawkermedia/amqp-client-provider/pull/9#discussion_r29174792>
> :
>
> >  			val result = Try(Await.result(producer.publish(msg.routingKey, Json.parse(msg.message)), republishTimeout))
> >  			result match {
> >  				case Success(_) => deleteMessageAndMatchingConfirm(msg, confs, transactional)
> > -				case Failure(ex) => logger.warn(s"""Couldn't resend message: $msg, ${ex.getMessage}""")
> > +				case Failure(ex: TimeoutException) =>
> > +					deleteMessageAndMatchingConfirm(msg, confs, transactional)
>
> If the publish times out, can we be sure that the new message will be
> saved to the msg store? What if there's a db connection problem?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/gawkermedia/amqp-client-provider/pull/9/files#r29174792>
> .
>